snuba/admin/views.py

```diff
@@ -278,6 +279,93 @@ def clickhouse_trace_query() -> Response:
         )
 
 
+@application.route("/clickhouse_querylog_query", methods=["POST"])
+def clickhouse_querylog_query() -> Response:
+    user = request.headers.get(USER_HEADER_KEY)
+    if not user:
+        return Response(
+            json.dumps({"error": "Unauthorized"}),
+            401,
+            {"Content-Type": "application/json"},
+        )
+    req = json.loads(request.data)
+    try:
+        raw_sql = req["sql"]
+    except KeyError as e:
+        return make_response(
+            jsonify(
+                {
+                    "error": {
+                        "type": "request",
+                        "message": f"Invalid request, missing key {e.args[0]}",
+                    }
+                }
+            ),
+            400,
+        )
+    try:
+        result = run_querylog_query(raw_sql, user)
+        rows = []
+        rows, columns = cast(List[List[str]], result.results), result.meta
+
+        if columns:
+            res = {}
+            res["column_names"] = [name for name, _ in columns]
+            res["rows"] = [[str(col) for col in row] for row in rows]
```

Comment:
Just followed an example endpoint which returns ClickHouse data:
https://github.com/getsentry/snuba/blob/85c38174832ecad29103b1daac697cd1a78d119f/snuba/admin/views.py#L205-L215

I assumed there is a reason for this, I don't have enough local data to cover some edge case this may be handling but I'll try testing without the cast.

Reply:
I don't think we want this with the querylog. The system queries have a limit to what they can return. The Querylog queries could be returning numbers, lists, booleans or whatever. Preserving that will make it easier to format in the front end.